meilisearch-http/src/backup.rs

```diff
@@ -0,0 +1,418 @@
+        }
+        Ok(())
+    })?;
+
+    Ok(())
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum BackupStatus {
+    Done,
+    Processing,
+    BackupProcessFailed,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct BackupInfo {
+    pub uid: String,
+    pub status: BackupStatus,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub error: Option<String>,
+}
+
+impl BackupInfo {
+    pub fn new(uid: String, status: BackupStatus) -> Self {
+        Self { uid, status, error: None }
+    }
+
+    pub fn with_error(mut self, error: String) -> Self {
+        self.status = BackupStatus::BackupProcessFailed;
+        self.error = Some(error);
+
+        self
+    }
+
+    pub fn backup_already_in_progress(&self) -> bool {
+        self.status == BackupStatus::Processing
+    }
+
+    pub fn get_current() -> Option<Self> {
+        BACKUP_INFO.lock().unwrap().clone()
+    }
+
+    pub fn set_current(&self) {
+        *BACKUP_INFO.lock().unwrap() = Some(self.clone());
+    }
+}
+
+/// Generate uid from creation date
+fn generate_uid() -> String {
+    Utc::now().format("%Y%m%d-%H%M%S").to_string()
+}
+
+/// Infer backup_folder from backup_uid
+pub fn compressed_backup_folder(backup_folder: &Path, backup_uid: &str) -> PathBuf {
+    backup_folder.join(format!("{}.tar.gz", backup_uid))
+}
+
+/// Write metadata in backup
+fn backup_metadata(data: &web::Data<Data>, folder_path: &Path, indexes: Vec<IndexResponse>) -> Result<(), Error> {
+    let (db_major, db_minor, db_patch) = data.db.version();
+    let metadata = BackupMetadata::new(indexes, format!("{}.{}.{}", db_major, db_minor, db_patch));
+
+    metadata.to_path(folder_path)
+}
+
+/// Export settings of provided index in backup
+fn backup_index_settings(data: &web::Data<Data>, reader: &MainReader, folder_path: &Path, index_uid: &str) -> Result<(), Error> {
+    let settings = crate::routes::setting::get_all_sync(data, reader, index_uid)?;
+
+    settings_to_path(&settings, folder_path)
+}
+
+/// Export updates of provided index in backup
+fn backup_index_updates(data: &web::Data<Data>, reader: &UpdateReader, folder_path: &Path, index_uid: &str) -> Result<(), Error> {
+    let updates_path = folder_path.join("updates.jsonl");
+    let updates = crate::routes::index::get_all_updates_status_sync(data, reader, index_uid)?;
+
+    let file = File::create(updates_path)?;
+
+    for update in updates {
+        serde_json::to_writer(&file, &update)?;
+        writeln!(&file)?;
+    }
+
+    Ok(())
+}
+
+/// Export documents of provided index in backup
+fn backup_index_documents(data: &web::Data<Data>, reader: &MainReader, folder_path: &Path, index_uid: &str) -> Result<(), Error> {
+    let documents_path = folder_path.join("documents.jsonl");
+    let file = File::create(documents_path)?;
+    let limit = 512;
```

Comment:
What is this arbitrary limit? It would be great to put it in the settings/args maybe, or at least move it as a constant a tthe top of this file with a better name.